src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+
+    const folder = targetUri ? vscode.workspace.getWorkspaceFolder(targetUri) : undefined
+
+    await new SamDebugConfigProvider(context).resolveDebugConfiguration(folder, finalConfig)
+}
+
+function getUriFromLaunchConfig(config: AwsSamDebuggerConfiguration): vscode.Uri | undefined {
+    let targetPath: string
+    if (isTemplateTargetProperties(config.invokeTarget)) {
+        targetPath = config.invokeTarget.templatePath
+    } else if (isCodeTargetProperties(config.invokeTarget)) {
+        targetPath = config.invokeTarget.projectRoot
+    } else {
+        // error
+        return undefined
+    }
+    if (path.isAbsolute(targetPath)) {
+        return vscode.Uri.file(targetPath)
+    }
+    const workspaceFolders = vscode.workspace.workspaceFolders || []
+    for (const workspaceFolder of workspaceFolders) {
+        const absolutePath = tryGetAbsolutePath(workspaceFolder, targetPath)
+        if (fs.pathExistsSync(absolutePath)) {
+            return vscode.Uri.file(absolutePath)
+        }
+    }
+
+    return undefined
+}
+
+function makeSampleRequestManifestResourceFetcher(): ResourceFetcher {
+    return new CompositeResourceFetcher(
+        new HttpResourceFetcher(sampleRequestManifestPath, { showUrl: true }),
+        new FileResourceFetcher(ext.manifestPaths.lambdaSampleRequests)
+    )
+}
+
+function getLaunchConfigQuickPickItems(launchConfig: LaunchConfiguration, uri: vscode.Uri): LaunchConfigPickItem[] {
+    const existingConfigs = launchConfig.getDebugConfigurations()
+    const samValidator = new DefaultAwsSamDebugConfigurationValidator(vscode.workspace.getWorkspaceFolder(uri))
+    return existingConfigs
+        .map((val, index) => {
+            return {
+                config: val,
+                index,
+            }
+        })
+        .filter(o => samValidator.validate((o.config as any) as AwsSamDebuggerConfiguration)?.isValid)
+        .map(val => {
+            return {
+                index: val.index,
+                label: val.config.name,
+                config: val.config as AwsSamDebuggerConfiguration,
+            }
+        })
+}
+
+export function finalizeConfig(config: AwsSamDebuggerConfiguration, name: string): AwsSamDebuggerConfiguration {
+    const newConfig = doTraverseAndPrune(config)
+    newConfig.name = name
+
+    if (isTemplateTargetProperties(config.invokeTarget)) {
+        newConfig.invokeTarget = {
+            target: config.invokeTarget.target,
+            logicalId: config.invokeTarget.logicalId,
+            templatePath: config.invokeTarget.templatePath,
+        }
+    } else if (isCodeTargetProperties(config.invokeTarget)) {
+        newConfig.invokeTarget = {
+            target: config.invokeTarget.target,
+            lambdaHandler: config.invokeTarget.lambdaHandler,
+            projectRoot: config.invokeTarget.projectRoot,
+        }
+    }
+
+    return newConfig
+}
+
+function doTraverseAndPrune(object: { [key: string]: any }): any | undefined {
```

Comment:
Likely belongs in `src/shared/utilities/collectionUtils.ts`.

deserves a docstring, something like
```suggestion
/** Removes empty fields from `object`. */
function doTraverseAndPrune(object: { [key: string]: any }): any | undefined {
```